app/presenters/queries/available_translations.rb

```diff
@@ -16,28 +16,58 @@ def translations
       attr_reader :content_id, :with_drafts, :expanded_translations
 
       def grouped_translations
-        Edition.with_document
-          .where("documents.content_id": content_id, state: state_fallback_order)
-          .pluck(:id, "documents.locale", :state)
-          .sort_by { |(_, _, state)| state_fallback_order.index(state.to_sym) }
-          .group_by { |(_, locale)| locale }
+        pluck_and_sort_editions(
+          filter_unpublishings(editions)
+        )
       end
 
       def expand_translation(id)
-        expansion_rules = LinkExpansion::Rules
-        web_item(id).select { |f| expansion_rules.expansion_fields(:available_translations).include?(f) }
+        web_item(id).select do |field|
+          expansion_rules
+            .expansion_fields(:available_translations)
+            .include?(field)
+        end
+      end
+
+      def expansion_rules
+        LinkExpansion::Rules
       end
 
       def web_item(id)
         Edition.find_by(id: id).to_h
       end
 
       def expanded_translations
-        @expanded_translations ||= grouped_translations.map { |_, (id)| expand_translation(id) }
+        @expanded_translations ||= grouped_translations.map do |_, (id)|
+          expand_translation(id)
+        end
       end
 
       def state_fallback_order
-        with_drafts ? %i[draft published] : %i[published]
+        return %i[draft published unpublished] if with_drafts
+        %i[published unpublished]
+      end
+
+      def editions
+        Edition
+          .with_document
+          .with_unpublishing
+          .where(
+            documents: { content_id: content_id },
+            state: state_fallback_order,
+          )
+      end
+
+      def filter_unpublishings(scope)
+        scope.where("
+          editions.state != 'unpublished' OR unpublishings.type = 'withdrawal'
+        ")
```

Comment:
Sure, I originally put it in the edition scope, but I found it looks a bit confusing why it was there. I opted for putting it in a separate method rather than having a comment, but maybe it is fine in the edition scope.